src/prr.rs

```diff
@@ -702,6 +742,128 @@ mod tests {
+            match var {
+                "GITHUB_TOKEN" => Ok("env_token".to_string()),
+                _ => Err(std::env::VarError::NotPresent),
+            }
+        };
+
+        let result = resolve_github_token(Some("config_token"), env_lookup).unwrap();
+        assert_eq!(result, "config_token");
+    }
+
+    #[test]
+    fn test_resolve_github_token_empty_config_token_falls_back_to_env() {
+        let env_lookup = |var: &str| -> Result<String, std::env::VarError> {
+            match var {
+                "GITHUB_TOKEN" => Ok("env_token".to_string()),
+                _ => Err(std::env::VarError::NotPresent),
+            }
+        };
+
+        let result = resolve_github_token(Some(""), env_lookup).unwrap();
+        assert_eq!(result, "env_token");
+    }
+
+    #[test]
+    fn test_resolve_github_token_env_var_precedence() {
+        let env_lookup = |var: &str| -> Result<String, std::env::VarError> {
+            match var {
+                "GH_TOKEN" => Ok("gh_token".to_string()),
+                "GITHUB_TOKEN" => Ok("github_token".to_string()),
+                _ => Err(std::env::VarError::NotPresent),
+            }
+        };
+
+        let result = resolve_github_token(None, env_lookup).unwrap();
+        // GH_TOKEN should have higher precedence
+        assert_eq!(result, "gh_token");
+    }
+
+    #[test]
+    fn test_resolve_github_token_empty_env_var_error() {
+        let env_lookup = |var: &str| -> Result<String, std::env::VarError> {
+            match var {
+                // Empty token
+                "GITHUB_TOKEN" => Ok("".to_string()),
+                _ => Err(std::env::VarError::NotPresent),
+            }
+        };
+
+        let result = resolve_github_token(None, env_lookup);
+        assert!(result.is_err());
+        let error_msg = result.err().unwrap().to_string();
+        assert!(error_msg.contains("Environment variable 'GITHUB_TOKEN' located but is empty"));
+    }
+
+    #[tokio::test]
+    async fn test_prr_with_missing_token_field_should_use_env_var() {
+        env::set_var("GITHUB_TOKEN", "env_var_token_value");
+
+        let gconfig = r#"
+            [prr]
+            workdir = "/tmp"
+        "#;
+
+        let (_prr, _dir) = config(gconfig, None);
+
+        env::remove_var("GITHUB_TOKEN");
+    }
+
+    #[tokio::test]
+    async fn test_env_var_fallback_to_token() {
+        let env_vars = [
+            "GH_TOKEN",
+            "GITHUB_TOKEN",
+            "GH_ENTERPRISE_TOKEN",
+            "GITHUB_ENTERPRISE_TOKEN",
+        ];
+        for var in &env_vars {
+            env::remove_var(var);
+        }
+
+        env::set_var("GITHUB_TOKEN", "github_token_fallback");
+
+        let gconfig = r#"
+            [prr]
+            token = "non_env_token"
+            workdir = "/tmp"
+        "#;
+
+        let result = config(gconfig, None);
+        assert!(result.0.config.prr.token == Some("non_env_token".to_string()));
+
+        env::remove_var("GITHUB_TOKEN");
+    }
```

Comment:
Hmm, these two tests will race with each other b/c they're modifying env vars, right? Maybe it's not necessary to have these - the above tests feel sufficient to me